Makefile

```diff
@@ -54,34 +54,42 @@ $(JAVA_CLIENT_JAR): dist/openapi-generator-cli.jar dist/java.swagger.json
+		--import-mappings ConfigMapKeySelector=io.kubernetes.client.openapi.models.V1ConfigMapKeySelector \
+		--import-mappings Container=io.kubernetes.client.openapi.models.V1Container \
+		--import-mappings ContainerPort=io.kubernetes.client.openapi.models.V1ContainerPort \
+		--import-mappings EnvFromSource=io.kubernetes.client.openapi.models.V1EnvFromSource \
+		--import-mappings EnvVar=io.kubernetes.client.openapi.models.V1EnvVar \
+		--import-mappings HostAlias=io.kubernetes.client.openapi.models.V1HostAlias \
+		--import-mappings Lifecycle=io.kubernetes.client.openapi.models.V1Lifecycle \
+		--import-mappings ListMeta=io.kubernetes.client.openapi.models.V1ListMeta \
+		--import-mappings LocalObjectReference=io.kubernetes.client.openapi.models.V1LocalObjectReference \
+		--import-mappings ObjectMeta=io.kubernetes.client.openapi.models.V1ObjectMeta \
+		--import-mappings ObjectReference=io.kubernetes.client.openapi.models.V1ObjectReference \
+		--import-mappings PersistentVolumeClaim=io.kubernetes.client.openapi.models.V1PersistentVolumeClaim \
+		--import-mappings PodDisruptionBudgetSpec=io.kubernetes.client.openapi.models.V1beta1PodDisruptionBudgetSpec \
+		--import-mappings PodDNSConfig=io.kubernetes.client.openapi.models.V1PodDNSConfig \
+		--import-mappings PodSecurityContext=io.kubernetes.client.openapi.models.V1PodSecurityContext \
+		--import-mappings Probe=io.kubernetes.client.openapi.models.V1Probe \
+		--import-mappings ResourceRequirements=io.kubernetes.client.openapi.models.V1ResourceRequirements \
+		--import-mappings SecretKeySelector=io.kubernetes.client.openapi.models.V1SecretKeySelector \
+		--import-mappings SecurityContext=io.kubernetes.client.openapi.models.V1SecurityContext \
+		--import-mappings Toleration=io.kubernetes.client.openapi.models.V1Toleration \
+		--import-mappings Volume=io.kubernetes.client.openapi.models.V1Volume \
+		--import-mappings VolumeDevice=io.kubernetes.client.openapi.models.V1VolumeDevice \
+		--import-mappings VolumeMount=io.kubernetes.client.openapi.models.V1VolumeMount \
 		--generate-alias-as-model \
 	# add the io.kubernetes:java-client to the deps
-	cd java && sed 's/<dependencies>/<dependencies><dependency><groupId>io.kubernetes<\/groupId><artifactId>client-java<\/artifactId><version>5.0.0<\/version><\/dependency>/g' pom.xml > tmp && mv tmp pom.xml
-    # I don't like these tests
+	cd java && sed -i 's|<dependencies>|<dependencies><dependency><groupId>io.kubernetes</groupId><artifactId>client-java</artifactId><version>9.0.2</version></dependency>|g' pom.xml
+	# implement KubernetesObject and KubernetesListObject on related classes
+	cd java/src/main/java/io/argoproj/workflow/models && \
+		sed -i 's|class Workflow {|class Workflow implements io.kubernetes.client.common.KubernetesObject {|g; s|private String kind|private String kind = "Workflow"|g; s|private String apiVersion|private String apiVersion = "argoproj.io/v1alpha1"|g' Workflow.java && \
```

Comment:
Ditto with this. Feels like the kind of thing that could benefit from a link to "why this kind of thing is necessary."

Reply:
this is to implement KubernetesObject and KubernetesListObject (line 84).  Do you mean expand on the need for implementing those?

Reply:
gotcha, makes sense. i'll add a link showing KubernetesObject extends KubernetesType, which requires `apiVersion` and `kind` (https://github.com/kubernetes-client/java/blob/b64ce915ad5964bebdc1d88b36dfe18b0892e574/kubernetes/src/main/java/io/kubernetes/client/common/KubernetesType.java#L22)

Reply:
Sweet, yeah, that’ll be perfect.